trie/node.go

```diff
@@ -48,7 +48,8 @@ import (
 	"github.com/ChainSafe/gossamer/common"
 )
 
-type node interface {
+// Node is the interface for trie methods
+type Node interface {
```

Comment:
this doesn't need to be exported

Reply:
the issue is that the type is being used as return value for public exported methods. We should either make it public or methods private.

Reply:
let's make the methods private

Reply:
roger